### DISCUSSION TITLE
API endpoint for listing games in a hub + list hubs that a game is in

### DISCUSSION THREAD

**Discussion by Bregann** [RetroAchievements/RAWeb#1229] [Ideas]
Hello,

I've got a couple suggestions around hubs via the API. Currently you can get the list of hubs by calling GetGameList however there's no way of using this to get the list of games that are in the hub from what I can tell. Would it be possible to get an API endpoint that lists the games in the hub from the ID? Similar to GetGame/GetGameExtended

Also, a 2nd suggestion around hubs. On the GetGameExtended endpoint would it be possible to return the hubs that the game is in? 

Thanks

**Comment by wescopeland** [comment]
~~I will add an endpoint for this once the testing story for the v1 API is figured out~~

**Reply by luchaos** [reply]
Some concerns to take into consideration: 

Game hubs were introduced by bastardizing the existing game alternatives feature - as that's the only thing we had available in the v1 schema.

Adding an API to this will make it substantially harder to cleaning up the schema and not using the game resource for hubs anymore. Game collections should not be a game resource.

V2's schema to eventually design game hubs and similar:
https://github.com/RetroAchievements/RAWeb/blob/master/database/migrations/platform/2018_10_02_000002_create_games_table.php#L71-L95

To prevent more work in the future I'd personally prefer to wait for that schema to be in place and in use - otherwise we create an API endpoint that requires additional data (reference to the original game ID) to be maintained.
